packages/code-gen/src/generator/types.js

```diff
@@ -263,6 +263,9 @@ export function generateTypeDefinition(
         if (Array.isArray(type.keys.oneOf)) {
           result += `{ [ key in `;
           result += generateTypeDefinition(context, type.keys, recurseSettings);
+        } else if (Array.isArray(type.keys.reference.oneOf)) {
```

Comment:
nit: reference can be undefined

eg: `T.generic("tmp").keys(T.string()).values(T.string()),`

results in 
![image](https://user-images.githubusercontent.com/42715253/122214306-3b532900-ceaa-11eb-8845-3053d05263ca.png)


```suggestion
        } else if (Array.isArray(type.keys.reference?.oneOf)) {
```

Reply:
aah oke, thanks!